internal/pkg/agent/application/actions/handlers/handler_action_unenroll.go

```diff
@@ -88,3 +135,30 @@ func (h *Unenroll) Handle(ctx context.Context, a fleetapi.Action, acker acker.Ac
 
 	return nil
 }
+
+func findMatchingUnitsByActionType(state state.State, typ string) ([]component.Component, []component.Unit) {
```

Comment:
This can be reused and can probably live outside of the unenroll source file.

Reply:
There is no other case for this at the moment. Once there is a need to use the same code in other places we can extract/refactor as needed.